docusaurus/docs/ui-extensions/create-an-extension-point.md

```diff
@@ -15,40 +15,285 @@ keywords:
+
+#### Return value
+
+The hooks return an object in the following format:
+
+```tsx
+// usePluginExtensions()
+{
+  isLoading: boolean;
+  extensions: Array<PluginExtensionLink | PluginExtensionComponent>
+}
+
+// usePluginLinkExtensions()
+{
+  isLoading: boolean;
+  extensions: PluginExtensionLink[]
+}
+
+// usePluginComponentExtensions()
+{
+  isLoading: boolean;
+  extensions: PluginExtensionComponent[]
+}
+```
+
+(For more information check the type definitions of [`PluginExtensionLink`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/pluginExtensions.ts#L27) and [`PluginExtensionComponent`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/pluginExtensions.ts#L35C13-L35C37).)
+
+#### Example - rendering link extensions (static context)
+
+The following example shows how to render a link component all link-type extensions that other plugins registered for the `plugins/another-app-plugin/menu` extension point ID.
+
+```tsx
+import { usePluginLinkExtensions } from '@grafana/runtime';
+
+// We are defining the `context` outside of the React component for performance reasons.
+// (In case we would declare it inside the component below, then it would result in a new object on every render,
+// which would unnecessarily trigger the `usePluginLinkExtensions()` hook.)
+const context = {
+  referenceId: '12345',
+  timeZone: 'UTC',
+};
+
+function AppMenuExtensionPoint() {
+  // This only returns type="link" extensions
+  const { extensions } = usePluginLinkExtensions({
+    extensionPointId: 'plugins/another-app-plugin/menu',
+    context,
+  });
+
+  if (extensions.length === 0) {
+    return null;
+  }
+
+  return (
+    <div>
+      {extensions.map((extension) => {
+        return (
+          <a href={extension.path} onClick={extension.onClick} title={extension.description} key={extension.key}>
+            {extension.title}
+          </a>
+        );
+      })}
+    </div>
+  );
+}
+```
+
+#### Example - rendering link extensions (dynamic context)
+
+The following example shows how to create the context object dynamically. Although this is a common practice, you should be aware that the `usePluginLinkExtensions()` hook will re-render in the following scenarios:
+
+- If the `context` object changes _(so the extensions can react to the context changes)_
+- If the extension-registry changes
+
+Be sure to only change the `context` object if its content changes; otherwise, you could create unnecessary re-renders. The following example shows how to approach these scenarios in a safe way:
+
+```tsx
+import { useMemo } from 'react';
+import { usePluginLinkExtensions } from '@grafana/runtime';
+
+function AppMenuExtensionPoint({ referenceId }) {
+  // Instead of just defining the object here (which would result in a new object on every render),
+  // we use `useMemo()` to only update the context object when its "dynamic" dependencies change.
```

Comment:
```suggestion
  // Instead of defining the object here (which would result in a new object on every render),
  // we use `useMemo()` to only update the context object when its "dynamic" dependencies change.
```